File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/invoice/presenter/InvoicePresenter.java
package org.mifos.mobilewallet.mifospay.invoice.presenter;

import android.net.Uri;

import org.mifos.mobilewallet.core.base.UseCase;
import org.mifos.mobilewallet.core.base.UseCaseHandler;
import org.mifos.mobilewallet.core.domain.usecase.invoice.FetchInvoice;
import org.mifos.mobilewallet.mifospay.base.BaseView;
import org.mifos.mobilewallet.mifospay.data.local.PreferencesHelper;
import org.mifos.mobilewallet.mifospay.invoice.InvoiceContract;

import javax.inject.Inject;

/**
 * Created by ankur on 07/June/2018
 */

public class InvoicePresenter implements InvoiceContract.InvoicePresenter {

    private final UseCaseHandler mUseCaseHandler;
    private final PreferencesHelper mPreferencesHelper;
    InvoiceContract.InvoiceView mInvoiceView;
    @Inject
    FetchInvoice fetchInvoiceUseCase;

    @Inject
    public InvoicePresenter(UseCaseHandler useCaseHandler, PreferencesHelper preferencesHelper) {
        mUseCaseHandler = useCaseHandler;
        mPreferencesHelper = preferencesHelper;
    }


    @Override
    public void attachView(BaseView baseView) {
        mInvoiceView = (InvoiceContract.InvoiceView) baseView;
        mInvoiceView.setPresenter(this);
    }

    @Override
    public void getInvoiceDetails(final Uri data) {

        mUseCaseHandler.execute(fetchInvoiceUseCase, new FetchInvoice.RequestValues(data),
                new UseCase.UseCaseCallback<FetchInvoice.ResponseValue>() {
                    @Override
                    public void onSuccess(FetchInvoice.ResponseValue response) {
                        mInvoiceView.showInvoiceDetails(response.getInvoices().get(0),
                                mPreferencesHelper.getFullName() + " "
                                        + mPreferencesHelper.getClientId(), data.toString());
                    }

                    @Override
                    public void onError(String message) {
                        mInvoiceView.showToast(message);
                    }
                });
    }
}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/invoice/presenter/InvoicePresenter.kt
package org.mifos.mobilewallet.mifospay.invoice.presenter

import android.net.Uri
import org.mifos.mobilewallet.core.base.UseCase.UseCaseCallback
import org.mifos.mobilewallet.core.base.UseCaseHandler
import org.mifos.mobilewallet.core.domain.usecase.invoice.FetchInvoice
import org.mifos.mobilewallet.mifospay.base.BaseView
import org.mifos.mobilewallet.datastore.PreferencesHelper
import org.mifos.mobilewallet.mifospay.invoice.InvoiceContract
import org.mifos.mobilewallet.mifospay.invoice.InvoiceContract.InvoiceView
import javax.inject.Inject

/**
 * Created by ankur on 07/June/2018
 */
class InvoicePresenter @Inject constructor(
    private val mUseCaseHandler: UseCaseHandler,
    private val mPreferencesHelper: PreferencesHelper,
    private val fetchInvoiceUseCase: FetchInvoice
) : InvoiceContract.InvoicePresenter {
    var mInvoiceView: InvoiceView? = null

    override fun attachView(baseView: BaseView<*>?) {
        mInvoiceView = baseView as InvoiceView?
        mInvoiceView!!.setPresenter(this)
    }

    override fun getInvoiceDetails(data: Uri?) {
        mUseCaseHandler.execute(fetchInvoiceUseCase, FetchInvoice.RequestValues(data),
            object : UseCaseCallback<FetchInvoice.ResponseValue> {
                override fun onSuccess(response: FetchInvoice.ResponseValue) {
                    mInvoiceView?.showInvoiceDetails(
                        response.invoices[0],
                        mPreferencesHelper.fullName + " "
                                + mPreferencesHelper.clientId, data.toString()
                    )
                }

                override fun onError(message: String) {
                    mInvoiceView!!.showToast(message)
                }
            })
    }
}